runtime/wasm/src/to_from/external.rs

```diff
@@ -324,6 +325,95 @@ impl ToAscObj<AscEthereumBlock> for EthereumBlockData {
+            difficulty: heap.asc_new(&BigInt::from_unsigned_u256(&self.difficulty)),
+            total_difficulty: heap.asc_new(&BigInt::from_unsigned_u256(&self.total_difficulty)),
+            size: self
+                .size
+                .map(|size| heap.asc_new(&BigInt::from_unsigned_u256(&size)))
+                .unwrap_or_else(|| AscPtr::null()),
+            transactions: heap.asc_new(self.transactions.as_slice()),
+        }
+    }
+}
+
+impl ToAscObj<AscFullEthereumBlockWithReceipts> for FullEthereumBlockDataWithReceipts {
+    fn to_asc_obj<H: AscHeap>(&self, heap: &mut H) -> AscFullEthereumBlockWithReceipts {
+        AscFullEthereumBlockWithReceipts {
+            hash: heap.asc_new(&self.hash),
+            parent_hash: heap.asc_new(&self.parent_hash),
+            uncles_hash: heap.asc_new(&self.uncles_hash),
+            author: heap.asc_new(&self.author),
+            state_root: heap.asc_new(&self.state_root),
+            transactions_root: heap.asc_new(&self.transactions_root),
+            receipts_root: heap.asc_new(&self.receipts_root),
+            number: heap.asc_new(&BigInt::from(self.number)),
+            gas_used: heap.asc_new(&BigInt::from_unsigned_u256(&self.gas_used)),
+            gas_limit: heap.asc_new(&BigInt::from_unsigned_u256(&self.gas_limit)),
+            timestamp: heap.asc_new(&BigInt::from_unsigned_u256(&self.timestamp)),
+            difficulty: heap.asc_new(&BigInt::from_unsigned_u256(&self.difficulty)),
+            total_difficulty: heap.asc_new(&BigInt::from_unsigned_u256(&self.total_difficulty)),
+            size: self
+                .size
+                .map(|size| heap.asc_new(&BigInt::from_unsigned_u256(&size)))
+                .unwrap_or_else(|| AscPtr::null()),
+            transaction_receipts: heap.asc_new(self.transaction_receipts.as_slice()),
+        }
+    }
+}
+
+impl ToAscObj<AscEthereumTransactionReceipt> for EthereumTransactionReceiptData {
+    fn to_asc_obj<H: AscHeap>(&self, heap: &mut H) -> AscEthereumTransactionReceipt {
+        AscEthereumTransactionReceipt {
+            //from receipts
+            hash: heap.asc_new(&self.hash),
+            index: heap.asc_new(&BigInt::from(self.index)),
+            cumulative_gas_used: heap
+                .asc_new(&BigInt::from_unsigned_u256(&self.cumulative_gas_used)),
+            gas_used: self
+                .gas_used
+                .map(|gas_used| heap.asc_new(&BigInt::from_unsigned_u256(&gas_used)))
+                .unwrap_or_else(|| AscPtr::null()),
+            contract_address: self
+                .contract_address
+                .map(|contract_address| heap.asc_new(&contract_address))
+                .unwrap_or_else(|| AscPtr::null()),
+            status: self
+                .status
+                .map(|status| heap.asc_new(&BigInt::from(status)))
+                .unwrap_or_else(|| AscPtr::null()),
+            root: self
```

Comment:
There's probably a helper trait/method that could be created to map an option to a value or null.

Reply:
Yeah I like that idea; this pattern is repeated all over the file. 
I'll add:
```rust
impl<C: AscType> From<Option<AscPtr<C>>> for AscPtr<C> {
    fn from(option: Option<AscPtr<C>>) -> Self {
        match option {
            Some(ptr) => ptr,
            None => AscPtr::null(),
        }
    }
}
```